app/views/markets/_form.html.erb

```diff
@@ -0,0 +1,21 @@
+<%= form_for @market do |f| %>
+  <%= f.label :name, "Market name:" %>
+  <%= f.text_field :name, required: :true %>
```

Comment:
You should validate on your models, too. Someone can modify an HTML form relatively easily and submit whatever they want, so it's important to protect and normalize your data on the model level, too. For instance, this field has the required attribute set to true, but I can inspect that element in a browser, remove that attribute, and then successfully create a market without a name.